app/controllers/miq_ae_tools_controller.rb

```diff
@@ -194,44 +194,55 @@ def review_import
     redirect_options = {:action => :review_git_import}
     git_url = params[:git_url]
-    verify_ssl = params[:git_verify_ssl] == "true" ? OpenSSL::SSL::VERIFY_PEER : OpenSSL::SSL::VERIFY_NONE 
+    verify_ssl = params[:git_verify_ssl] == "true" ? OpenSSL::SSL::VERIFY_PEER : OpenSSL::SSL::VERIFY_NONE
+    new_git_repo = nil
 
     if git_url.blank?
       add_flash(_("Please provide a valid git URL"), :error)
     elsif !GitBasedDomainImportService.available?
       add_flash(_("Please enable the git owner role in order to import git repositories"), :error)
     else
-      if GitRepository.exists?(:url => git_url)
-        flash_message = _("This repository has been used previously for imports; If you use the same domain it will get deleted and recreated")
-        status = :warning
-      else
-        flash_message = _("Successfully found git repository, please choose a branch or tag")
-        status = :success
-      end
+      begin
+        git_repo = GitRepository.find_by(:url => git_url)
+        unless git_repo
+          git_repo = GitRepository.create!(:url => git_url)
+          new_git_repo = git_repo
+        end
+        git_repo.update_attributes(:verify_ssl => verify_ssl)
+        if params[:git_username] && params[:git_password]
+          git_repo.update_authentication(:values => {:userid   => params[:git_username],
+                                                     :password => params[:git_password]})
+        end
 
-      git_repo = GitRepository.create(:url => git_url, :verify_ssl => verify_ssl)
-      git_repo.update_authentication(:values => {:userid => params[:git_username], :password => params[:git_password]})
-      task_options = {
-        :action => "Retrieve git repository",
-        :userid => current_user.userid
-      }
-      queue_options = {
-        :class_name  => "GitRepository",
-        :method_name => "refresh",
-        :instance_id => git_repo.id,
-        :role        => "git_owner",
-        :args        => []
-      }
-
-      task_id = MiqTask.generic_action_with_callback(task_options, queue_options)
-      MiqTask.wait_for_taskid(task_id)
-
-      add_flash(flash_message, status)
-      branch_names = git_repo.git_branches.collect(&:name)
-      tag_names = git_repo.git_tags.collect(&:name)
-      redirect_options[:git_branches] = branch_names.to_json
-      redirect_options[:git_tags] = tag_names.to_json
-      redirect_options[:git_repo_id] = git_repo.id
+        task_options = {
+          :action => "Retrieve git repository",
+          :userid => current_user.userid
+        }
+        queue_options = {
+          :class_name  => "GitRepository",
+          :method_name => "refresh",
+          :instance_id => git_repo.id,
+          :role        => "git_owner",
+          :args        => []
+        }
+
+        task_id = MiqTask.generic_action_with_callback(task_options, queue_options)
+        MiqTask.wait_for_taskid(task_id)
+
+        task = MiqTask.find(task_id)
```

Comment:
You can remove this line.  `MiqTask.wait_for_taskid` returns the `task` object which has been reloaded.

Just use: `task = MiqTask.wait_for_taskid(task_id)`